xbmc/games/addons/GameClient.cpp

```diff
@@ -694,3 +696,34 @@ bool CGameClient::cb_input_event(KODI_HANDLE kodiInstance, const game_input_even
 
   return gameClient->Input().ReceiveInputEvent(*event);
 }
+
+void CGameClient::ParseLibretroName(const std::string& addonName,
```

Comment:
what about
```suggestion
std::pair<std::string, std::string> CGameClient::ParseLibretroName(const std::string& addonName,
```

Then you can use:
```
[m_emulatorName, m_platforms] = ParseLibretroName(Name());
```

Reply:
Me likey

Reply:
@lrusak Can you help with your suggestion? I make the following changes:

```patch
--- a/xbmc/games/addons/GameClient.h
+++ b/xbmc/games/addons/GameClient.h
@@ -18,6 +18,7 @@
 #include <set>
 #include <stdint.h>
 #include <string>
+#include <utility>
 
 class CFileItem;
 
@@ -214,7 +215,23 @@ private:
   static bool cb_input_event(KODI_HANDLE kodiInstance, const game_input_event* event);
   //@}
 
-  static void ParseLibretroName(const std::string& addonName, std::string& emulatorName, std::string& platforms);
+  /*!
+   * \brief Parse the name of a libretro game add-on into its emulator name
+   * and platforms
+   *
+   * \param addonName The name of the add-on, e.g. "Nintendo - SNES / SFC (Snes9x 2002)"
+   *
+   * \return A tuple of two strings:
+   *   - first:  the emulator name, e.g. "Snes9x 2002"
+   *   - second: the platforms, e.g. "Nintendo - SNES / SFC"
+   *
+   * For cores that don't emulate a platform, such as 2048 with the add-on name
+   * "2048", then the emulator name will be the add-on name and platforms
+   * will be empty, e.g.:
+   *   - first:  "2048"
+   *   - second: ""
+   */
+  static std::pair<std::string, std::string> ParseLibretroName(const std::string& addonName);
```

and

```patch
--- a/xbmc/games/addons/GameClient.cpp
+++ b/xbmc/games/addons/GameClient.cpp
@@ -107,7 +107,7 @@ CGameClient::CGameClient(const ADDON::AddonInfoPtr& addonInfo)
   m_bSupportsStandalone =
       addonInfo->Type(AddonType::GAMEDLL)->GetValue(GAME_PROPERTY_SUPPORTS_STANDALONE).asBoolean();
 
-  ParseLibretroName(Name(), m_emulatorName, m_platforms);
+  [m_emulatorName, m_platforms] = ParseLibretroName(Name());
 }
```

But when I compile, I get the error:

```
GameClient.cpp: In constructor ‘KODI::GAME::CGameClient::CGameClient(const AddonInfoPtr&)’:
GameClient.cpp:110:4: error: capture of non-variable ‘KODI::GAME::CGameClient::m_emulatorName’
  110 |   [m_emulatorName, m_platforms] = ParseLibretroName(Name());
      |    ^~~~~~~~~~~~~~
In file included from GameClient.cpp:9:
GameClient.h:244:15: note: ‘std::string KODI::GAME::CGameClient::m_emulatorName’ declared here
  244 |   std::string m_emulatorName;
      |               ^~~~~~~~~~~~~~
```

Does this not work for member variables?

Reply:
Try
```
std::tie(m_emulatorName, m_platforms) = ParseLibretroName(Name());
```